File: src/main/java/net/csibio/aird/compressor/intcomp/Empty.java
package net.csibio.aird.compressor.intcomp;

import net.csibio.aird.enums.IntCompType;

public class Empty implements IntComp {

    @Override
    public String getName() {
        return IntCompType.Empty.getName();
    }

    @Override
    public int[] encode(int[] uncompressed) {
        return uncompressed;
    }

    @Override
    public int[] decode(int[] compressed) {
        return compressed;
    }
}

File: PythonSDK/Compressor/IntComp/Empty.py
from Enums.IntCompType import IntCompType


class Empty:

    def getName(self):
        return IntCompType.Empty

    def encode(self, input):
        return input

    def decode(self, input):
        return input

    def decode(self, input, offset, length):
        return input[offset, offset + length]
